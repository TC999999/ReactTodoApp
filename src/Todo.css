.complete {
  text-decoration: line-through;
}

.btn {
  margin: 5px;
}
