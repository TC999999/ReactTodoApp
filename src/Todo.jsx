import "./Todo.css";

const Todo = ({ id, todo, complete, delTodo, compTodo, showEditForm }) => {
  const classes = `task ${complete ? "complete" : ""}`;
  const compMessage = () => {
    return complete == true ? "Mark As Incomplete" : "Mark As Complete";
  };

  return (
    <div className="todo">
      <div className="todo-item" id={id}>
        <p className={classes}>- {todo}</p>
        <button className="btn" onClick={compTodo}>
          {compMessage()}
        </button>
        <button className="btn" onClick={delTodo}>
          X
        </button>
        <button className="btn" onClick={showEditForm}>
          Edit Task
        </button>
      </div>
    </div>
  );
};

export default Todo;
