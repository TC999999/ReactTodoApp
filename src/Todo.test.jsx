import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo.jsx";
import { vi } from "vitest";

it("renders without crashing", function () {
  render(<Todo />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("should show a button that says 'Mark As Complete' if complete is false'", function () {
  const { queryByText } = render(<Todo complete={false} />);
  const btn = queryByText("Mark As Complete");
  expect(btn).toBeInTheDocument();
});

it("should show a button that says 'Mark As Incomplete' if complete is true'", function () {
  const { queryByText } = render(<Todo complete={true} />);
  const btn = queryByText("Mark As Incomplete");
  expect(btn).toBeInTheDocument();
});

it("should give any todos with a complete attribute a class of 'complete'", function () {
  const { queryByText } = render(<Todo todo="test-task" complete={true} />);
  const task = queryByText("- test-task");
  expect(task.className).toBe("task complete");
});

it("should call a function to show the edit form when edit button is pressed", function () {
  const compTodoMock = vi.fn();
  const { queryByText } = render(
    <Todo complete={false} compTodo={compTodoMock} />
  );
  expect(compTodoMock).not.toHaveBeenCalled();
  const btn = queryByText("Mark As Complete");
  fireEvent.click(btn);
  expect(compTodoMock).toHaveBeenCalled();
});

it("should call a function to show the edit form when edit button is pressed", function () {
  const showEditFormMock = vi.fn();
  const { queryByText } = render(<Todo showEditForm={showEditFormMock} />);
  expect(showEditFormMock).not.toHaveBeenCalled();
  const btn = queryByText("Edit Task");
  fireEvent.click(btn);
  expect(showEditFormMock).toHaveBeenCalled();
});

it("should call a function to show to delete a task when delete button is pressed", function () {
  const delTodoMock = vi.fn();
  const { queryByText } = render(<Todo delTodo={delTodoMock} />);
  expect(delTodoMock).not.toHaveBeenCalled();
  const btn = queryByText("X");
  fireEvent.click(btn);
  expect(delTodoMock).toHaveBeenCalled();
});
