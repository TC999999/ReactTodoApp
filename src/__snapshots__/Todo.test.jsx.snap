// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo"
  >
    <div
      class="todo-item"
    >
      <p
        class="task "
      >
        - 
      </p>
      <button
        class="btn"
      >
        Mark As Complete
      </button>
      <button
        class="btn"
      >
        X
      </button>
      <button
        class="btn"
      >
        Edit Task
      </button>
    </div>
  </div>
</DocumentFragment>
`;
